### DISCUSSION TITLE
Building Portable Version for Linux

### DISCUSSION THREAD

**Discussion by Rimpire** [eteran/nedit-ng#161] [Q&A]
This is more of a question rather than an issue.
I am not able to compile nedit-ng on all of Linux systems I am using, since some are missing dependencies and I have no root access on them.
Is there a way to build a portable executable with all dependencies included in the build folder ? 
I am very novice to cmake searching around I tried to use the pointers [from here](https://stackoverflow.com/questions/36932916/build-a-portable-executable-with-its-dependencies) but with no luck.

**Comment by eteran** [comment]
Definitely should be doable, I'll look into what it takes!

**Comment by Rimpire** [comment]
Any luck looking into this ?

**Comment by bogen85** [comment]
@Rimpire  > Any luck looking into this ?

You mean a statically linked build?
Are runc or crun installed on any of those systems?

**Comment by eteran** [comment]
Supposedly, AppImage will do the trick, gonna look into it!

**Comment by bogen85** [comment]
Yeah, that should work. Shared libraries not on the target system have to be bundled with the executable (similar to how DLLs are packaged with executable files on Windows).  [AppImage uses LD_LIBRARY_PATH](https://github.com/AppImage/AppImageKit/search?q=LD_LIBRARY_PATH&unscoped_q=LD_LIBRARY_PATH)  Which in general should work, as that is not easy to disable the use of.

**Comment by bogen85** [comment]
See [AppImage uses FUSE](https://github.com/AppImage/AppImageKit/search?q=fuse&unscoped_q=fuse) but that can be worked around if the user can't use it on the target system.

**Comment by eteran** [comment]
It'll be hard to make perfect, but it looks like it'll work for most circumstances. I'll try to make a build target for it 👍
